src/update_engine/payload_processor.cc

```diff
@@ -170,8 +173,30 @@ ActionExitCode PayloadProcessor::LoadManifest() {
 ActionExitCode PayloadProcessor::PerformOperation() {
   DCHECK(next_operation_num_ < num_total_operations_);
 
-  const InstallOperation *op = &manifest_.partition_operations(next_operation_num_);
-  DeltaPerformer *performer = &partition_performer_;
+  const size_t total_partition_operations = manifest_.partition_operations_size();
+  const InstallOperation *op = nullptr;
+  DeltaPerformer *performer = nullptr;
+
+  if (next_operation_num_ < total_partition_operations) {
+    op = &manifest_.partition_operations(next_operation_num_);
+    performer = &partition_performer_;
+  } else {
```

Comment:
Because that is how we construct the payloads. The operations must be processed in the same order as the data they reference in the payload since it is processing the payload in a streaming manner. Obviously in the manifest_.procedures() list has built-in ordering so that works out fine but the partition_operations list predates that so we just have to pick the ordering between the two. Also we can't actually order them the other way around without breaking compatibility with old versions of update_engine which expect the partition data to be first.